src/modules/auto-google-map.js

```diff
@@ -0,0 +1,172 @@
+		var lat = element.attr('data-lat');
+		var lng = element.attr('data-lng');
+		var iconSrc = element.attr('data-icon');
+		var iconWidth = parseInt(element.attr('data-icon-width'), 10) || DEFAULT_MARKER_WIDTH;
+		var iconHeight = parseInt(element.attr('data-icon-height'), 10) || DEFAULT_MARKER_HEIGHT;
+
+		if (!!App.device.internetexplorer) {
+			iconSrc = null;
+		}
+
+		var config = {
+			id: element.attr('data-poi-id'),
+			LatLng: new window.google.maps.LatLng(lat, lng),
+			content: element.html(),
+			iconImage: {
+				src: iconSrc,
+				size: new google.maps.Size(iconWidth, iconHeight),
+				scaledSize: new google.maps.Size(iconWidth, iconHeight),
+				p1: {
+					x: 0,
+					y: 0
+				},
+				p2: {
+					x: iconWidth / 2,
+					y: iconHeight
+				}
+			},
+			zIndex: 50
+		};
+
+		if (!iconSrc) {
+			delete config.iconImage;
+		}
+
+		var marker = mapComponent.addMarker(config);
+
+		element.data('markerRef', marker);
+	};
+
+	var initAllMap = function () {
+
+		if (!ready) {
+			return;
+		}
+
+		scope.find(sels.ctn).each(function () {
+			var t = $(this);
+
+			if (!!t.find(sels.map).data('googleMapComp')) {
+				return;
+			}
+
+			var zoom = !!t.attr('data-zoom') ? parseInt(t.attr('data-zoom'), 10) : DEFAULT_ZOOM;
+			var lat = t.attr('data-lat');
+			var lng = t.attr('data-lng');
+			var markers = t.find(sels.marker);
+			var styles = t.find(sels.styles).html();
+
+			if ((!lat || !lng) && !!markers.length) {
+				lat = markers.first().attr('data-lat');
+				lng = markers.first().attr('data-lng');
+			}
+
+			if (!!styles) {
+				try {
+					styles = JSON.parse(styles);
+				} catch (error) {
+					App.log({
+						me: 'Auto Google Maps',
+						args: error,
+						fx: 'error'
+					});
+				}
+			}
+
+			if (!!lat && !!lng) {
+				var comp = App.components.create('googleMap', {
+					zoom: zoom,
+					center: {
+						latitude: lat,
+						longitude: lng
+					},
+					selectorCtn: sels.map,
+					styles: styles || undefined,
+					zoomControl: true,
+					panControl: true,
+					streetViewControl: false,
+					mapTypeControl: false,
+					scrollwheel: false,
+					mapTypeId: 'roadmap',
+					draggable: true,
+					markerCustomAction: function (event, marker) {
+						if (!!marker.attr('data-url')) {
+							var win = window.open(marker.attr('data-url'), '_blank');
+							win.focus();
+						}
+					},
+					afterCreate: function () {
+						//Add pins
+						markers.each(function () {
+							createMarker($(this), comp);
+						});
+					}
+				});
+
+				// Store comp
+				t.data('googleMapComp', comp);
+				comp.init(t);
+			}
+
+		});
+	};
+
+	var onPageEnter = function (key, data) {
+		scope = $(data.page.key());
+		initAllMap();
+		ready = true;
+	};
+
+	var onArticleEntering = function () {
+		initAllMap();
+	};
+
+	var onSiteLoaded = function () {
+		initAllMap();
+		ready = true;
+	};
+
+	var actions = function () {
+		return {
+			site: {
+				loaded: onSiteLoaded
+			},
+			page: {
```

Comment:
We should destroy before leaving, if the page gets removed.

Reply:
The api does not include a destory method, removing the dom will do the job.

Reply:
> The api does not include a destory method,

You mean Gmaps API ?

I that case, we could at least remove the event handlers before removing from the dom ?

see https://stackoverflow.com/questions/10485582/what-is-the-proper-way-to-destroy-a-map-instance

Reply:
@fhamon ping!

Reply:
> The official answer is you don't.

Reply:
Yeah I can't agree with that...